packages/react/src/AvatarStack/AvatarStack.tsx

```diff
@@ -303,20 +371,29 @@ const AvatarStack = ({
   }
 
   const avatarStackSx = merge<BetterCssProperties | BetterSystemStyleObject>(
-    getResponsiveAvatarSizeStyles(),
+    !enabled && getResponsiveAvatarSizeStyles(),
     sxProp as SxProp,
   )
 
   return (
-    <AvatarStackWrapper count={count} className={wrapperClassNames} sx={avatarStackSx}>
-      <Box
-        className={bodyClassNames}
-        tabIndex={!hasInteractiveChildren && !disableExpand ? 0 : undefined}
-        ref={stackContainer}
+    <AvatarStackWrapper
+      count={enabled ? undefined : count}
+      data-avatar-count={enabled ? (count > 3 ? '3+' : count) : undefined}
+      data-align-right={enabled && alignRight ? '' : undefined}
+      data-responsive={enabled && (!size || isResponsiveValue(size)) ? '' : undefined}
+      className={clsx(wrapperClassNames, {[classes.AvatarStack]: enabled})}
+      // @ts-ignore - it's not allowing CSS properties here
+      style={enabled ? (getResponsiveAvatarSizeStyles() as React.CSSProperties) : undefined}
```

Comment:
I think you can get rid of the ts-ignore and the as  React.CSSProperties since it is being aliased within the function

```suggestion
      style={enabled ? getResponsiveAvatarSizeStyles() : undefined}
```